test/e2e/common_suite_test.go

```diff
@@ -715,7 +739,33 @@ func doTestCreatePeerPodWithAuthenticatedImage(t *testing.T, assert CloudAssert)
 	expectedAuthStatus := "Completed"
 	secretData := map[string][]byte{v1.DockerConfigJsonKey: authfile}
 	secret := newSecret(namespace, secretName, secretData, v1.SecretTypeDockerConfigJson)
-	imageName := os.Getenv("REGISTRY_IMAGE")
+	imageName := os.Getenv("AUTHENTICATED_REGISTRY_IMAGE")
+	pod := newPod(namespace, podName, podName, imageName, withRestartPolicy(v1.RestartPolicyNever), withImagePullSecrets(secretName))
+	newTestCase(t, "ValidAuthImagePeerPod", assert, "Peer pod with Authenticated Image with Valid Credentials has been created").withSecret(secret).withPod(pod).withAuthenticatedImage().withAuthImageStatus(expectedAuthStatus).withCustomPodState(v1.PodSucceeded).run()
+}
+
+func doTestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials(t *testing.T, assert CloudAssert) {
```

Comment:
Hey Sudharshan, did you see this/have any thoughts?

Reply:
Yes,
I tried by passing an empty space for `REGISTRY_CREDENTIAL_ENCODED` , it returned an error `illegal base64 data at input byte`. I also added the case in current [test condition](https://github.com/sudharshanibm3/cloud-api-adaptor/blob/e2e-test-auth-images-valid/test/e2e/common_suite_test.go#L524) itself.

We can add another test-case with no credentials, but it will also act with the same logic used by `doTestCreatePeerPodWithAuthenticatedImageWithInvalidCredentials `

Reply:
Sorry, what I was more thinking of was trying to pull the `AUTHENTICATED_REGISTRY_IMAGE`, but with no auth.json secret created at all. This would help us catch the issue if the registry image didn't actually required authentication and stop us potentially have a false positive on the `TestCreatePeerPodWithAuthenticatedImagewithValidCredentials/ValidAuthImagePeerPod_test`. Does that make sense, or have I explained it badly?